chain-extensions/pallet-assets/src/lib.rs

```diff
@@ -0,0 +1,411 @@
+            }
+            AssetsFunc::TransferApproved => {
+                let (origin, id, owner, destination, amount): (
+                    Origin,
+                    <T as pallet_assets::Config>::AssetId,
+                    T::AccountId,
+                    T::AccountId,
+                    T::Balance,
+                ) = env.read_as()?;
+
+                let base_weight = <T as pallet_assets::Config>::WeightInfo::transfer_approved();
+                env.charge_weight(base_weight)?;
+
+                let runtime_origin = RawOrigin::Signed(match origin {
+                    Origin::Caller => env.ext().caller().clone(),
+                    Origin::Address => env.ext().address().clone(),
+                });
+
+                let call_result = pallet_assets::Pallet::<T>::transfer_approved(
+                    runtime_origin.into(),
+                    id.into(),
+                    owner.into(),
+                    destination.into(),
+                    amount,
+                );
+                return match call_result {
+                    Err(e) => {
+                        let mapped_error = Outcome::from(e);
+                        Ok(RetVal::Converging(mapped_error as u32))
+                    }
+                    Ok(_) => Ok(RetVal::Converging(Outcome::Success as u32)),
+                };
+            }
+            AssetsFunc::SetMetadata => {
+                let (origin, id, name, symbol, decimals): (
+                    Origin,
+                    <T as pallet_assets::Config>::AssetId,
+                    Vec<u8>,
+                    Vec<u8>,
+                    u8,
+                ) = env.read_as_unbounded(env.in_len())?;
+
+                let base_weight = <T as pallet_assets::Config>::WeightInfo::set_metadata(
+                    name.len() as u32,
+                    symbol.len() as u32,
+                );
+                env.charge_weight(base_weight)?;
+
+                let runtime_origin = RawOrigin::Signed(match origin {
+                    Origin::Caller => env.ext().caller().clone(),
+                    Origin::Address => env.ext().address().clone(),
+                });
+
+                let call_result = pallet_assets::Pallet::<T>::set_metadata(
+                    runtime_origin.into(),
+                    id.into(),
+                    name,
+                    symbol,
+                    decimals,
+                );
+                return match call_result {
+                    Err(e) => {
+                        let mapped_error = Outcome::from(e);
+                        Ok(RetVal::Converging(mapped_error as u32))
+                    }
+                    Ok(_) => Ok(RetVal::Converging(Outcome::Success as u32)),
+                };
+            }
+            AssetsFunc::MetadataName => {
+                let id: <T as pallet_assets::Config>::AssetId = env.read_as()?;
+
+                let base_weight = <T as frame_system::Config>::DbWeight::get().reads(1);
```

Comment:
Better to have it in same manner with other weight getters, magic numbers is not best practice.